### PR TITLE
Set the default LT version to the new LT version

### PR DESCRIPTION
**Pull request by olemarkus**
Needed for Karpenter, which will always use the default version of a Launch Template.

### CODE REVIEW

File: upup/pkg/fi/cloudup/awstasks/launchtemplate_target_api.go [hunk 1]
```diff
@@ -164,8 +166,19 @@ func (t *LaunchTemplate) RenderAWS(c *awsup.AWSAPITarget, a, e, changes *LaunchT
 			LaunchTemplateName: t.Name,
 			LaunchTemplateData: data,
 		}
-		if _, err = c.Cloud.EC2().CreateLaunchTemplateVersion(input); err != nil {
+		if version, err := c.Cloud.EC2().CreateLaunchTemplateVersion(input); err != nil {
 			return fmt.Errorf("error creating LaunchTemplateVersion: %v", err)
+		} else {
```
Comment (hakman) [inline upup/pkg/fi/cloudup/awstasks/launchtemplate_target_api.go:171]: Please keep the same style.
```suggestion
		version, err := c.Cloud.EC2().CreateLaunchTemplateVersion(input)
		if err != nil {
			return fmt.Errorf("error creating LaunchTemplateVersion: %v", err)
		}
		...
```

File: upup/pkg/fi/cloudup/awstasks/launchtemplate_target_api.go [hunk 2]
```diff
@@ -164,8 +166,19 @@ func (t *LaunchTemplate) RenderAWS(c *awsup.AWSAPITarget, a, e, changes *LaunchT
...
+			newDefault := strconv.FormatInt(*version.LaunchTemplateVersion.VersionNumber, 10)
+			_, err := c.Cloud.EC2().ModifyLaunchTemplate(
+				&ec2.ModifyLaunchTemplateInput{
+					DefaultVersion:   &newDefault,
+					LaunchTemplateId: version.LaunchTemplateVersion.LaunchTemplateId,
+				},
+			)
```
Comment (hakman) [inline upup/pkg/fi/cloudup/awstasks/launchtemplate_target_api.go:178]: Could we do this just for Karpenter? Users already complain about the amount of API calls.

**Review by hakman — CHANGES_REQUESTED** [review]

File: upup/pkg/fi/cloudup/awstasks/launchtemplate_target_api.go (hunk 2 above)
Comment (olemarkus) [reply upup/pkg/fi/cloudup/awstasks/launchtemplate_target_api.go:178]: Users should not hit rate limits of this call at least. It's mostly non-mutating call that is a challenge, but these are separate buckets.

LTs do not know about karpenter, and I can't think of a reason why they would. I also can't think of how to inject such a value in a good way.

Comment (hakman) [reply upup/pkg/fi/cloudup/awstasks/launchtemplate_target_api.go:178]: OK, we can go with it as is and optimize later.

File: upup/pkg/fi/cloudup/awstasks/launchtemplate_target_api.go (hunk 1 above)
Comment (hakman) [reply upup/pkg/fi/cloudup/awstasks/launchtemplate_target_api.go:171]: I really dislike the `if err {} else {}` pattern, it's hard to follow, but can be changed later.

**Review by hakman — APPROVED** [review]

File: upup/pkg/fi/cloudup/awstasks/launchtemplate_target_api.go (hunk 1 above)
Comment (johngmyers) [reply upup/pkg/fi/cloudup/awstasks/launchtemplate_target_api.go:171]: Concur. There should not be an `else` after an if block that always returns.
